server/src/main/java/org/candlepin/dto/api/v1/ProductCertificateDTO.java

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright (c) 2009 - 2012 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.dto.api.v1;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.annotations.ApiModel;
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+import org.candlepin.common.jackson.HateoasInclude;
+
+/**
+ * The ProductCertificateDTO is a DTO representing product certificates presented to the API.
+ */
+@ApiModel(parent = TimestampedCandlepinDTO.class, description = "DTO representing a product certificate")
+public class ProductCertificateDTO extends TimestampedCandlepinDTO<ProductCertificateDTO> {
+
+    private static final long serialVersionUID = 1L;
+
+    protected String id;
+    protected String key;
+    protected String cert;
+    private ProductDTO product;
+
+    /**
+     * Initializes a new ProductCertificateDTO instance with null values.
+     */
+    public ProductCertificateDTO() {
+        // Intentionally left empty
+    }
+
+    /**
+     * Initializes a new ProductCertificateDTO instance which is a shallow copy of the provided
+     * source entity.
+     *
+     * @param source
+     *  The source entity to copy
+     */
+    public ProductCertificateDTO(ProductCertificateDTO source) {
+        super(source);
+    }
+
+    /**
+     * Retrieves the id field of this ProductCertificateDTO object.
+     *
+     * @return the id field of this ProductCertificateDTO object.
+     */
+    @HateoasInclude
+    public String getId() {
+        return this.id;
+    }
+
+    /**
+     * Sets the id to set on this ProductCertificateDTO object.
+     *
+     * @param id the id to set on this ProductCertificateDTO object.
+     *
+     * @return a reference to this ProductCertificateDTO object.
+     */
+    public ProductCertificateDTO setId(String id) {
```

Comment:
This should also be marked with `@JsonIgnore` to make sure it isn't set during deserialization, since the model entity didn't have a setter either.